packages/react-dev-utils/WebpackDevServerUtils.js

```diff
@@ -103,6 +103,7 @@ function printInstructions(appName, urls, useYarn) {
 
 function createCompiler({
   appName,
+  appPath,
```

Comment:
Maybe we should give this a default value of `''`?

Reply:
if we give it a default value of '' then it somewhat becomes purposeless; `appPath` is passed around so we can strip the prefix path from lots of places.  eg `C:/source/my-app/src/something.tsx` -> `src/something.tsx`  

I'm beginning to have mixed feelings about the path prefix stripping actually.  Whilst it initially seemed like a good idea and reduced "noise" that rather enforces a certain way of working with your codebase.  Let me explain:

Consider the following (not unusual) project structure:

```
package.json
src/
    client/
        src/
            package.json
            // ...
    server/
        src
            package.json
            // ...
```

With this structure it's quite common to want to open up in the root of that structure and work from there, hacking on both `client` and `server` at the same time.    You may fire up the VS Code and, run a `yarn start:client` script which behind the scenes does this `cd src/client && yarn start`

*Without* the stripping path prefixes approach this works fine.  The paths that appear in the console are clickable as they are complete paths.  However with path stripping in effect VS Code doesn't know where to go when you're working in the root of the project.  Where is this `src/` you speak of?  It's confused and not unreasonably.

We lose click to go to error in console *entirely* in this scenario due to path prefix stripping.  You can work around it by opening up only in the root of the create-react-app application and working there.  Then it'll be fine.  But this feels like we're rather forcing a way of working upon users.  

What do you think?  My feeling is that users would probably be better served if we dropped the path prefix stripping.

Reply:
Thanks for the detailed explanation.

I agree. All of our path stripping solutions so far have had caveats and been somewhat complicated. Maybe the best first implementation is to use full paths. If we can come up with a robust solution in the future, then it will be an improvement.

Reply:
Solid - I'll make it so 👍